### PR TITLE
8280047: Broken link to Swing Connection document from javax.swing package docs

### PR DESCRIPTION
**Pull request by prsadhuk**
Fixed the broken link with appropriate link to swing document.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280047](https://bugs.openjdk.java.net/browse/JDK-8280047): Broken link to Swing Connection document from javax.swing package docs

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - Author) ⚠️ Review applies to efcf062cd2556b2ae0ec1d7ab141b15966a54c56
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7104/head:pull/7104` \
`$ git checkout pull/7104`

Update a local copy of the PR: \
`$ git checkout pull/7104` \
`$ git pull https://git.openjdk.java.net/jdk pull/7104/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7104`

View PR using the GUI difftool: \
`$ git pr show -t 7104`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7104.diff">https://git.openjdk.java.net/jdk/pull/7104.diff</a>

</details>

### CODE REVIEW

**Comment by prsadhuk** [comment]
It seems "Swing connection" used to point to group of swing articles but is removed long back 
https://community.oracle.com/tech/developers/discussion/1214597/where-is-swing-connection
I guess it's better to point to Swing architecture overview instead.

File: src/java.desktop/share/classes/javax/swing/package-info.java [hunk 1]
```diff
@@ -132,8 +132,8 @@
  * For overviews, tutorials, examples, guides, and other documentation,
  * please see:
  * <ul>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/tech/articles-jsp-139072.html"
- *     target="_top">The Swing Connection</a></li>
+ *     <li><a href=""https://www.oracle.com/java/technologies/a-swing-architecture.html"
```
Comment (kevinrushforth) [inline src/java.desktop/share/classes/javax/swing/package-info.java:135]: Typo: you have an extra `"` on this line.

**Review by kevinrushforth — COMMENTED** [review]
This looks like a good replacement, but you have a typo in the HTML link. Once fixed, please run `make docs` and check the generated html page.

**Comment by kevinrushforth** [comment]
> ...you have a typo in the HTML link.

I see that the GitHub actions run failed for the same reason.

**Review by kevinrushforth — APPROVED** [review]

File: src/java.desktop/share/classes/javax/swing/package-info.java [hunk 2]
```diff
@@ -132,8 +132,8 @@
  * For overviews, tutorials, examples, guides, and other documentation,
  * please see:
  * <ul>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/tech/articles-jsp-139072.html"
- *     target="_top">The Swing Connection</a></li>
+ *     <li><a href="https://www.oracle.com/java/technologies/a-swing-architecture.html"
+ *     target="_top">A Swing Architecture Overview</a></li>
  *     <li><a href="https://docs.oracle.com/javase/tutorial/"
  *     target="_top">The Java Tutorial</a></li>
  *     <li><a href="http://www.oracle.com/technetwork/java/javase/training/index.html"
```
Comment (aivanov-jdk) [inline src/java.desktop/share/classes/javax/swing/package-info.java:139]: While links are updated, shall the following links be updated too?
The links redirect to another page which have different titles.

“Online Training at the Java Developer Connection” now points to “Java SE Training and Certification” page: https://www.oracle.com/java/technologies/javase/training-support.html

“Java Foundation Classes (JFC) home page” leads to “Java SE Desktop Overview”: https://www.oracle.com/java/technologies/javase/desktop-overview.html

**Review by aivanov-jdk — APPROVED** [review]

File: src/java.desktop/share/classes/javax/swing/package-info.java (hunk 2 above)
Comment (prsadhuk) [reply src/java.desktop/share/classes/javax/swing/package-info.java]: fixed other links

File: src/java.desktop/share/classes/javax/swing/package-info.java [hunk 3]
```diff
@@ -132,14 +132,14 @@
...
  *     <li><a href="https://docs.oracle.com/javase/tutorial/"
  *     target="_top">The Java Tutorial</a></li>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/training/index.html"
- *     target="_top">Online Training</a>
+ *     <li><a href="https://www.oracle.com/java/technologies/javase/training-support.html"
+ *     target="_top">Java SE Training and Certification</a>
  *     at the Java Developer Connection <sup>SM</sup></li>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/tech/index-jsp-142216.html"
+ *     <li><a href="https://www.oracle.com/java/technologies/java-foundation-classes.html"
  *     target="_top">Java Foundation Classes (JFC)</a> home page</li>
```
Comment (aivanov-jdk) [inline src/java.desktop/share/classes/javax/swing/package-info.java:143]: Hm, I think this page is too outdated by now. It talks about JDK 1.4.0. It's completely irrelevant for JDK 19, is it?

Shall this link be dropped altogether?

Comment (prsadhuk) [reply src/java.desktop/share/classes/javax/swing/package-info.java:143]: Maybe we can use https://www.oracle.com/java/technologies/foundation-classes-faq.html

Comment (prsadhuk) [reply src/java.desktop/share/classes/javax/swing/package-info.java:143]: Yes, the page looks outdated.. I don't think any new content has been added post jdk1.4.2.
I can remove the link. 
In fact, I think the "JavaSE Training" link also is not relevant in this swing package, maybe it should be put in overall parent overview page.

Comment (aivanov-jdk) [reply src/java.desktop/share/classes/javax/swing/package-info.java]: > In fact, I think the "JavaSE Training" link also is not relevant in this swing package, maybe it should be put in overall parent overview page.

Yeah, I agree it's not relevant for Swing.

Do you refer to [Java SE at a Glance](https://www.oracle.com/java/technologies/java-se-glance.html)?

Comment (aivanov-jdk) [reply src/java.desktop/share/classes/javax/swing/package-info.java]: [Java SE Desktop Overview](https://www.oracle.com/java/technologies/javase/desktop-overview.html) is relevant as the overview of all desktop related technologies in Java. Yet it's not specifically about Swing. As such, this page fits for java.desktop module description better.

Comment (aivanov-jdk) [reply src/java.desktop/share/classes/javax/swing/package-info.java]: > I guess in same way, it's more relevant to https://docs.oracle.com/en/java/javase/17/docs/api/index.html page

Yep!

**Review by aivanov-jdk — APPROVED** [review]

File: src/java.desktop/share/classes/javax/swing/package-info.java [hunk 4]
```diff
@@ -132,15 +132,13 @@
...
- *     target="_top">The Swing Connection</a></li>
+ *     <li><a href="https://www.oracle.com/java/technologies/a-swing-architecture.html"
+ *     target="_top">A Swing Architecture Overview</a></li>
  *     <li><a href="https://docs.oracle.com/javase/tutorial/"
  *     target="_top">The Java Tutorial</a></li>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/training/index.html"
- *     target="_top">Online Training</a>
+ *     <li><a href="https://www.oracle.com/java/technologies/javase/training-support.html"
+ *     target="_top">Java SE Training and Certification</a>
  *     at the Java Developer Connection <sup>SM</sup></li>
```
Comment (aivanov-jdk) [inline src/java.desktop/share/classes/javax/swing/package-info.java:141]: I don't have a strong opinion whether this link should be here. Maybe not, as you mentioned.

If it's kept, I think the part `at the Java Developer Connection <sup>SM</sup>` should be removed.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]
Shall the copyright year be updated?

File: src/java.desktop/share/classes/javax/swing/package-info.java [hunk 5]
```diff
@@ -132,15 +132,12 @@
...
+ *     target="_top">A Swing Architecture Overview</a></li>
  *     <li><a href="https://docs.oracle.com/javase/tutorial/"
  *     target="_top">The Java Tutorial</a></li>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/training/index.html"
- *     target="_top">Online Training</a>
- *     at the Java Developer Connection <sup>SM</sup></li>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/tech/index-jsp-142216.html"
- *     target="_top">Java Foundation Classes (JFC)</a> home page</li>
+ *     <li><a href="https://www.oracle.com/java/technologies/javase/training-support.html"
+ *     target="_top">Java SE Training and Certification</a>
```
Comment (aivanov-jdk) [inline src/java.desktop/share/classes/javax/swing/package-info.java:140]:
```suggestion
 *     target="_top">Java SE Training and Certification</a></li>
```
The closing `</li>` is not required but it's used in other `<li>` elements in this list.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

File: src/java.desktop/share/classes/javax/swing/package-info.java (hunk 5 above)
Comment (prsadhuk) [reply src/java.desktop/share/classes/javax/swing/package-info.java]: done

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**